### DISCUSSION TITLE
How to provide your own CBC binaries

### DISCUSSION THREAD

**Discussion by JeroenGar** [coin-or/python-mip#280] [Suggestions]
**Background:**
When using the 1.13.0 release, my code runs fine. 
However when using the current version on master with new CBC binaries, CBC crashes (after the timelimit is reached but before returning to python).

In order to investigate this issue I would like to test my code with more different versions of CBC.
It is clear to me that the binaries are located in python-mip\mip\libraries. With a single "cbc-c-linux-x86-64.so" of around 20MB for linux.
However, when downloading the compiled releases from [https://github.com/coin-or/Cbc/releases](https://github.com/coin-or/Cbc/releases) I can't seem to find a large monolithic .so file anywhere in the archive.

**Question:**
Where can I find instructions on how to use custom binaries for python-mip?

**Comment by jurasofish** [comment]
a couple of things that might help you 
- You can set the PMIP_CBC_LIBRARY env var to point to the binary you want to use.
- Stable versions of CBC seem to not work (https://github.com/coin-or/python-mip/issues/165) so you will probably want to compile CBC master from source (e.g. https://github.com/coin-or/python-mip/issues/165#issuecomment-1058946026)

**Reply by sebheger** [reply]
The cbc stable version do not work because the c-interface necessary for the python-mip package are only fully included in the master branch of cbc. And cbc release are yet not build from master. So you need to build/compile your own cbc and the link it the way @jurasofish described.

**Reply by JeroenGar** [reply]
Thanks @jurasofish and @sebheger, I am now able to use my own binaries on Linux using coinbrew.
